extensions/answer_summarizers/models.py

```diff
@@ -177,15 +220,44 @@ class AnswerFrequencies(BaseCalculation):
     submitted).
     """
 
-    def calculate_from_state_answers_dict(self, state_answers_dict):
+    def calculate_from_state_answers_dict(
+        self, state_answers_dict: stats_domain.StateAnswersDict
+    ) -> stats_domain.StateAnswersCalcOutput:
         """Computes the number of occurrences of each answer, and returns a list
         of dicts; each dict has keys 'answer' and 'frequency'.
 
         This method is run from within the context of a MapReduce job.
+
+        Args:
+            state_answers_dict: dict. A dict containing state answers and
+                exploration information such as:
+                * exploration_id: id of the exploration.
+                * exploration_version: Specific version of the exploration or
+                    VERSION_ALL is used if answers are aggregated across
+                    multiple versions.
+                * state_name: Name of the state.
+                * interaction_id: id of the interaction.
+                * submitted_answer_list: A list of submitted answers.
+
+        Returns:
+            stats_domain.StateAnswersCalcOutput. A calculation output object
+            containing the answers' frequencies.
+
+        Raises:
+            Exception. Answers of linear interactions should not be present
+                while calculating the answers' frequencies.
         """
         answer_dicts = state_answers_dict['submitted_answer_list']
+        answer_list = []
+        for answer_dict in answer_dicts:
+            if answer_dict['answer'] is None:
+                raise Exception(
```

Comment:
How are we guaranteeing this? Should we also have a clause earlier on to say something like "assert interaction ID is not linear"?

Reply:
> How are we guaranteeing this?

In the whole codebase including tests, I didn't find any case where we are calculating answer frequencies for Linera interactions ( by linear interaction I mean `continue` interaction ) https://github.com/oppia/oppia/blob/741cd122e84ac041ac5e52585e038ae3d86a6cdb/core/feconf.py#L829

If I did mistake in understanding something I'm happy to remove this raise exception :)

Reply:
Also, at this place, we are validating that linear interactions should have None answers only https://github.com/oppia/oppia/blob/741cd122e84ac041ac5e52585e038ae3d86a6cdb/core/domain/stats_domain.py#L1608

Reply:
ditto for other cases!

Reply:
Added new check PTAL!

Reply:
Ditto for all other cases !